cli/aws_provider_patch.go

```diff
@@ -0,0 +1,131 @@
+		}
+
+		updatedTerraformFileContents, codeWasUpdated, err := patchAwsProviderInTerraformCode(originalTerraformFileContents, terraformFile, terragruntOptions.AwsProviderPatchOverrides)
+		if err != nil {
+			return err
+		}
+
+		if codeWasUpdated {
+			terragruntOptions.Logger.Printf("Patching AWS provider in %s", terraformFile)
+			if err := util.WriteFileWithSamePermissions(terraformFile, terraformFile, []byte(updatedTerraformFileContents)); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// findAllTerraformFiles returns all Terraform source files within the modules being used by this Terragrunt
+// configuration. To be more specific, it only returns the source files downloaded for module "xxx" { ... } blocks into
+// the .terraform/modules folder; it does NOT return Terraform files for the top-level (AKA "root") module.
+//
+// NOTE: this method only supports *.tf files right now. Terraform code defined in *.json files is not currently
+// supported.d
+func findAllTerraformFilesInModules(terragruntOptions *options.TerragruntOptions) ([]string, error) {
+	modulesPath := util.JoinPath(terragruntOptions.DataDir(), "modules")
+
+	if !util.FileExists(modulesPath) {
+		return nil, nil
+	}
+
+	// Ideally, we'd use a builin Go library like filepath.Glob here, but per https://github.com/golang/go/issues/11862,
+	// the current go implementation doesn't support treating ** as zero or more directories, just zero or one.
+	// So we use a third-party library.
+	matches, err := zglob.Glob(fmt.Sprintf("%s/**/*.tf", modulesPath))
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	return matches, nil
+}
+
+// patchAwsProviderInTerraformCode looks for provider "aws" { ... } blocks in the given Terraform code and overwrites
+// the attributes in those provider blocks with the given attributes. It returns the new Terraform code and a boolean
+// true if that code was updated.
+//
+// For example, if you passed in the following Terraform code:
+//
+// provider "aws" {
+//    region = var.aws_region
+// }
+//
+// And you set attributesToOverride to map[string]string{"region": "us-east-1"}, then this method will return:
+//
+// provider "aws" {
+//    region = "us-east-1"
+// }
+//
+// This is a temporary workaround for a Terraform bug (https://github.com/hashicorp/terraform/issues/13018) where
+// any dynamic values in nested provider blocks are not handled correctly when you call 'terraform import', so by
+// temporarily hard-coding them, we can allow 'import' to work.
+func patchAwsProviderInTerraformCode(terraformCode string, terraformFilePath string, attributesToOverride map[string]string) (string, bool, error) {
+	hclFile, err := hclwrite.ParseConfig([]byte(terraformCode), terraformFilePath, hcl.InitialPos)
+	if err != nil {
+		return "", false, errors.WithStackTrace(err)
+	}
+
+	codeWasUpdated := false
+
+	for _, block := range hclFile.Body().Blocks() {
+		tokens := block.BuildTokens(nil)
+		if len(tokens) > 4 {
```

Comment:
If you switch to the newer HCL2 library (`"github.com/hashicorp/hcl/v2"`), you should have access to `block.Type()` and `block.Labels`, which should simplify this to:

```
if block.Type() == "provider" && len(block.Labels()) == 1 && block.Labels()[0] == "aws" {
```

Reply:
Whhhhhaaaat? Will try now, thanks! 😁

Reply:
Well goddamn. It worked. Nice one Yori! https://github.com/gruntwork-io/terragrunt/pull/1326/commits/bdd6eed5b941e31a365774ca0bf40170f4c92f90